src/webots/gui/WbGuiApplication.cpp

```diff
@@ -193,110 +194,164 @@ void WbGuiApplication::parseStreamArguments(const QString &streamArguments) {
       else
         mTaskArguments.clear();
       mTask = UPDATE_PROTO_CACHE;
-    } else if (arg.startsWith("--update-world"))
+    } else if (subCommand == "update-world" && parser.isSet(updateWorldOption)) {
+      args.pop_front();
       mTask = UPDATE_WORLD;
-    else if (arg == "--enable-x3d-meta-file-export")
+    } else if (subCommand == "enable-x3d-meta-file-export" && parser.isSet(x3DMetaFileExportOption)) {
+      args.pop_front();
       WbWorld::enableX3DMetaFileExport();
-    else if (arg.startsWith("--stream")) {
-      stream = true;
-      QString serverArgument;
-      int equalCharacterIndex = arg.indexOf('=');
-      if (equalCharacterIndex != -1) {
-        serverArgument = arg.mid(equalCharacterIndex + 1);
-        // remove starting/trailing double quotes
-        if (serverArgument.startsWith('"'))
-          serverArgument = serverArgument.right(serverArgument.size() - 1);
-        if (serverArgument.endsWith('"'))
-          serverArgument = serverArgument.left(serverArgument.size() - 1);
+    } else if (subCommand == "stream" && parser.isSet(streamOption)) {
+      args.pop_front();
+      QStringList streamArgument = parser.values(streamOption);
+      if (streamArgument.size() != 1) {
+        fprintf(stderr, "%s\n",
+                qPrintable(QCoreApplication::translate(
+                  "main", "webots: invalid option : '--log-performance': log file path is missing.")));
+        parser.showHelp(1);
       }
-      parseStreamArguments(serverArgument);
-    } else if (arg == "--stdout")
+      stream = true;
+      parseStreamArguments(streamArgument[0]);
+    } else if (subCommand == "stdout" && parser.isSet(stdoutOption)) {
+      args.pop_front();
       WbLog::enableStdOutRedirectToTerminal();
-    else if (arg == "--stderr")
+    } else if (subCommand == "stderr" && parser.isSet(stderrOption)) {
+      args.pop_front();
       WbLog::enableStdErrRedirectToTerminal();
-    else if (arg.startsWith("--log-performance")) {
-      int equalCharacterIndex = arg.indexOf('=');
-      if (equalCharacterIndex != -1) {
-        QString logArgument = arg.mid(equalCharacterIndex + 1);
-        // remove starting/trailing double quotes
-        if (logArgument.startsWith('"'))
-          logArgument = logArgument.right(logArgument.size() - 1);
-        if (logArgument.endsWith('"'))
-          logArgument = logArgument.left(logArgument.size() - 1);
-
-        if (logArgument.contains(",")) {
-          QStringList argumentsList = logArgument.split(",");
-          WbPerformanceLog::createInstance(argumentsList[0], argumentsList[1].trimmed().toInt());
-        } else
-          WbPerformanceLog::createInstance(logArgument);
-
+    } else if (subCommand == "log-performance" && parser.isSet(logPerfomanceOption)) {
+      QStringList logArgument = parser.values(logPerfomanceOption);
+      if (logArgument[0].contains(",")) {
+        QStringList argumentsList = logArgument[0].split(",");
```

Comment:
We try to use `const` everywhere it is possible to enhance readability and make the code more robust:
```suggestion
        const QStringList argumentsList = logArgument[0].split(",");
```